src/common_i2c.hpp

```diff
@@ -0,0 +1,78 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ * Copyright (C) 2022 YADRO.
+ */
+
+#pragma once
+
+#include <string>
+
+extern "C"
+{
+#include <i2c/smbus.h>
+}
+
+/**
+ * @class i2cDev
+ *
+ * This class implements low level communication with I2C device
+ */
+class i2cDev
+{
+  public:
+    /**
+     * @brief Constructor
+     *
+     * The constructor initializes communication descriptor and test if device
+     * accessibly.
+     * @param[in] devPath - I2C bus device file path (e.g. "/dev/i2c-20")
+     * @param[in] addr - 7-bit I2C device address
+     * @param[in] usePEC - If we should use PEC for communication
+     */
+    i2cDev(std::string devPath, int addr, bool usePEC = false);
+
+    /**
+     * @brief Destructor
+     *
+     * The destructor closes communication descriptor
+     */
+    ~i2cDev();
+
+    /**
+     * @brief If communication descriptor initialized successfully
+     *
+     * @return true if ok, false if any error appears during initialization
+     */
+    bool isOk()
```

Comment:
I prefer to make the such method const.
Some cases it allows to avoid unnecessary copying of the object.
But in this project this is not relevant and may be left as is.